src/tlskdf.rs

```diff
@@ -210,6 +267,11 @@ impl Mechanism for TLSPRFMechanism {
             return Err(CKR_MECHANISM_INVALID)?;
         }
 
+        #[cfg(feature = "fips")]
```

Comment:
I think this can be moved inside of TLSKDFOperation::new() (as the very first check) so you do not have to repeat it three times.

Reply:
You can also just do:
```
if TLS_PRF_SELFTEST.result != CKR_OK {
    return Err(TLS_PRF_SELFTEST.result)?;
}
```

Reply:
I was thinking about that, but did not want to indent the whole function. Will figure out something.

Reply:
no re-indenting necessary ?

Reply:
Doing it in `TLSKDFOperation()` won' t cover the `TLSMACOperation` then. I can move it into these two, but then we are just on 2 from 3 and one level deeper in the call stack.

It could be moved also into the `TLSPRF::new()`, but I was worried that it would be too deep to easily understand.

Reply:
Looks like we do not have any test coverage for the `CKM_TLS12_MAC` if I see right. That would be another TODO/next issue.

Reply:
Done.